lib/chkhash.c

```diff
@@ -0,0 +1,64 @@
+#include <regex.h>
+#include <string.h>
+#include <stdbool.h>
+#include <stddef.h>
```

Comment:
Please include `<config.h>` (separate from the rest).  Also, please sort them alphabetically.  Also, please include `"chkhash.h"`.

```c
#include <config.h>

#include "chkhash.h"

#include <regex.h>
#include <stdbool.h>
#include <stddef.h>
#include <string.h>
```